.github/workflows/publish-npm.yml

```diff
@@ -0,0 +1,62 @@
+name: Publish to NPM
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "21"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install Rust
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          source $HOME/.cargo/env
+
+      - name: Install Solana CLI
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/v1.18.15/install)"
+          export PATH="/home/runner/.local/share/solana/install/active_release/bin:$PATH"
+          solana --version
+
+      - name: Install Anchor CLI
+        run: |
+          cargo install --git https://github.com/coral-xyz/anchor --tag v0.30.0 anchor-cli --locked
+          anchor --version
+
+      - name: Build
+        run: anchor build
+
+      - name: Prepare IDL Files
+        run: |
+          mv target/idl ./
+
+      - name: Determine NPM Tag
+        id: determine-npm-tag
+        run: |
+          VERSION_TAG=${GITHUB_REF#refs/tags/v}
+          if [[ $VERSION_TAG == *"-"* ]]; then
+            echo ::set-output name=NPM_TAG::${VERSION_TAG#*-}
+          else
+            echo ::set-output name=NPM_TAG::latest
+          fi
+        env:
+          GITHUB_REF: ${{ github.ref }}
```

Comment:
This is deprecated by GitHub, please consider using:
```
steps:
  - name: Set the foo value
    id: foo_value
    run: |
      echo "foo=bar" >> "$GITHUB_ENV"
```

Reply:
I'm sorry, but this has never worked for me. I propose we improve this in a follow-up PR.

Reply:
There's a working example in a workflow on this repository:
```
        - name: Record Rust version
          run: echo "RUST=$(rustc --version)" >> "$GITHUB_ENV"
        - name: Upload to codecov.io
          uses: codecov/codecov-action@v4
          with:
            fail_ci_if_error: true
            token: ${{ secrets.CODECOV_TOKEN }}
            env_vars: OS,RUST
```

Reply:
bed715c